cmd/minishift/cmd/services/systray.go

```diff
@@ -0,0 +1,44 @@
+// +build !systemtray
+
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"runtime"
+
+	"github.com/anjannath/systray"
+	"github.com/minishift/minishift/pkg/minishift/systemtray"
+	"github.com/spf13/cobra"
+)
+
+// systrayCmd represents the systemtray command
+var systrayServiceCmd = &cobra.Command{
+	Use:   "systray",
+	Short: "Run minishift systemtray",
+	Long:  "Run a systemtray in the notification area of top bar/start menu",
+	Run: func(cmd *cobra.Command, args []string) {
+		systray.Run(systemtray.OnReady, systemtray.OnExit)
+	},
+	Hidden: true,
```

Comment:
@anjannath if we are exposing the `service` command then why this subcommand we are hiding?

Reply:
@praveenkumar This command `minishift services systray` do not put the tray to background, we call this command from `minishift services systemtray` after setting required flags for the process to make it a background process. So we don't need to expose the `systray` command to users.

Reply:
@praveenkumar Yes that is correct.

Reply:
I agree with @gbraad , this solution does not scale.